Krok Addict Species Name Fix

MadMedicineMan opened the pull request:
## About The Pull Request
Tweaks #50949. Krokodil Addicts were incorrectly displayed as "Humans" in medical scanners leading some players of non-human chars to believe they were turned into humans. Now, getting addicted to Krok only leads to visual change, as only the respective limbs_id is applied, with a say_mod instead of a zombie tounge. Thus, characters retain their species specific properties.

Krok Addict species is still in the code, but is only used to fetch the appropriate limbs_id/say_mod.

PR now provides a proc that only changes sprite to any given species, instead of mutating the player, with a toggable say_mod. (set_limb_id())

**Thanks a bunch @XDTM for helping a baby coder, would have never found the mistake witout you!**

## Why It's Good For The Game

- Fixes a consistency issue. (Krok Addicts should be less likely to be mistaken for real zombies)
- Adds an easy way to apply visual changes without mutating chars to a different species

## Changelog
:cl:
rcsdel: Krok Addiction does not change species anymore, only applies visual change
tweak: Changed krokodil addict species to be more consistent 
/:cl:

MadMedicineMan committed:
Changed krok addict species name, added alert

MadMedicineMan commented:
Very tiny edit, for getting into and exploring byond/tg code and getting started with git. Re-PR because I git-derped with the first one and other files were changed.

XDTM commented:
Krokodil addicts being a separate species is more of a crutch than an intentional decision. The initial implementation only wanted to change the skin sprite to the zombie one (to represent the skin falling off) and species was the simplest way to do it. Ideally the species would be removed in favor of manually changing the human species' icon without changing species altogether.

After all, it doesn't make sense for a human to no longer be human when they lose their skin.

MadMedicineMan commented:
@XDTM Im with you on that, and I think I have an idea on how to fix that. However, im unable to set the limb id via a similar method that is used to apply species changes.

XDTM commented:
There's no real need for a method, in theory you should be able to simply set `target_mob.dna.species.limbs_id = "zombie"` as long as you call `update_icon() `on it afterwards.

It will reset if the mob changes species, but it does that now anyway so it's not a big deal.

MadMedicineMan commented:
Darn, i might be overthinking and overengineering a way small problem.

MadMedicineMan commented:
So what ever way im trying, it starts turing the sprite invisible, and for the life of me i cant figure it out.

XDTM commented:
Ah, i forgot, while humans use greyscaled icons, zombies have a non-grayscaled static icon set, and they are stored in separate files.

You should set `use_skintones` to false on the species to make it use the right file.

Lizards and ethereals have MUTCOLORS and DYNCOLORS respectively which result in the same issue, so you'll also need to do `species_traits -= MUTCOLORS` and `species_traits -= DYNCOLORS`

MadMedicineMan committed:
Function for setting limbs_id, tweaks to krok addict species

Added a function similar to set_species that only sets and applies the limbs_id, resulting in only visual change, with optional say_mod change to a given species, instead of mutating to it. Added fix for transparent textures when switching to zombie limbs. Replaced zombie tongue in krok addicts with say mod, to fix consistency that they arent real Z's.

MadMedicineMan commented:
@XDTM Thanks a bunch, I finally got it to work. The new proc is the result of not being able to fix it the hacky way before i knew what was the problem, and I didnt want to chuck hours of work out the window a second time. 
You`re great!

XDTM commented on code/datums/dna.dm:
```diff
@@ -323,6 +323,40 @@
+		if(new_limb_id.limbs_id == "zombie")
+			dna.species.use_skintones = FALSE //prevent sprite turning transparent
+			dna.species.species_traits -= MUTCOLORS
+			dna.species.species_traits -= DYNCOLORS
```
This will cause issues whenever the limbs_id does not use the same color system as the original species. Trying to support all the different possibilities will be complicated if you intend to do it, since for example you could have a species using skintones morphing into a species with MUTCOLORS.

To be honest, if i were doing the PR, i'd avoid making this into a function altogether, since it's not a common occurrence, and if it was it would be better handled as a more radical rework.

MadMedicineMan replied:
Yeah, youre right, I gave it some thought and will redo it as a single fix just for that one case

ExcessiveUseOfCobblestone commented:
can we not take out the species component by letting you set a forced string in lieu of a species for the first argument?

MadMedicineMan commented:
This should in theory also work, so basically supply a string and use this to set the limp id? 

The problem im having currently is this: when trying to revert to just a one-time check (addicted to krok? then hard set the limb id and species trait flags and update the sprite), then sprites again turn invisible, even though it worked when called by the proc i wrote for this PR. And currently RL is sapping all my time so i cant fix it currently (even though its possibly a minor problem)

ExcessiveUseOfCobblestone commented:
you should be able to use either since I can see a usecase for both, you just turn the '.limbs_id' on 338 to a variable that is either built off the arg directly (if it is a string) or built off the '.limbs_id' if it is a species type.

MadMedicineMan commented:
I currently do not have the time to tinker with it, Im very sorry, life came back at me. So I will close this PR for now, maybe i can fix it some time. Id still be glad to add to this community and amazing game, thanks for all the input!
